tasks/main.yml

```
@@ -1,2 +1,95 @@
 ---
+- block:
+  - name: Determine latest GitHub release (local)
+    delegate_to: localhost
+    become: false
+    uri:
+      url: "https://api.github.com/repos/prometheus/node_exporter/releases/latest"
+      body_format: json
+    register: _github_release
+    until: _github_release.status == 200
+    retries: 5
+
+  - name: Set node_exporter_version
+    set_fact:
+      node_exporter_version: "{{ _github_release.json.tag_name
+        | regex_replace('^v?(.*)$', '\\1') }}"
+  when: (node_exporter_version is not defined) or (node_exporter_version == 'latest')
+  run_once: true
+
+- name: Get systemd version
+  command: systemctl --version
+  changed_when: false
+  check_mode: false
+  register: __systemd_version
+  tags:
+    - skip_ansible_lint
+
+- name: Set systemd version fact
+  set_fact:
+    node_exporter_systemd_version: "{{ __systemd_version.stdout_lines[0] | regex_replace('^systemd\\s(\\d+).*$', '\\1') }}"
+
+- name: create cache directory
+  file:
+    path: /var/cache/node_exporter/{{ node_exporter_version }}
+    state: directory
+
+- name: Download node_exporter and verify SHA256 checksum
```

```suggestion
- name: Download node_exporter
```
I'd say the hash verification is an implementation detail of the download step